### ISSUE TITLE
rpc error: code = 2 desc = oci runtime error: exec failed: container_linux.go:1153: sending signal 0 to pid 27592 caused "permission denied"

### ISSUE THREAD

**Issue by Jehanramadhan** [moby/moby#33646]
**Description**
Previously i never get this,
But suddenly today I get this error, I've tried by adding the parameter "sudo", but still the result is same.
I also have to re-install my docker, still the error appears when I run a command such as: docker stop, exec docker, etc.

**Steps to reproduce the issue:**
1. docker start <container_name>
2.docker exec -it <container_name> bash

**Describe the results you received:**
rpc error: code = 2 desc = oci runtime error: exec failed: container_linux.go:1153: sending signal 0 to pid 27592 caused "permission denied"

**Describe the results you expected:**
in bash of my docker container

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.03.1-ce-rc1
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   3476dbf
 Built:        Wed Mar 15 20:48:41 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.1-ce-rc1
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   3476dbf
 Built:        Wed Mar 15 20:48:41 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Client:
 Version:      17.03.1-ce-rc1
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   3476dbf
 Built:        Wed Mar 15 20:48:41 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.1-ce-rc1
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   3476dbf
 Built:        Wed Mar 15 20:48:41 2017
 OS/Arch:      linux/amd64
 Experimental: false
jehan [~] Do > docker info
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 2
Server Version: 17.03.1-ce-rc1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 54
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.19.0-32-generic
Operating System: Ubuntu 14.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.529 GiB
Name: ThinkPad
ID: TD5A:66WD:5E4L:QWRZ:CIBC:JHVE:TAVR:JUXY:6YXX:3OYW:GKNB:VMJS
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: jehanramadhan
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 192.168.0.167:5000
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by VoteForNavalny** [comment]
The same here.

> docker version 
> Client:
>  Version:      17.03.1-ce
>  API version:  1.27
>  Go version:   go1.7.5
>  Git commit:   c6d412e
>  Built:        Mon Mar 27 17:14:09 2017
>  OS/Arch:      linux/amd64
> 
> Server:
>  Version:      17.03.1-ce
>  API version:  1.27 (minimum version 1.12)
>  Go version:   go1.7.5
>  Git commit:   c6d412e
>  Built:        Mon Mar 27 17:14:09 2017
>  OS/Arch:      linux/amd64
>  Experimental: false

> docker ps
> CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS               NAMES
> fec7a9230e6a        mysql:latest        "docker-entrypoint..."   12 minutes ago      Up 12 minutes       3306/tcp            rdm_db_1
> 

> 
> d exec -it fec7a9230e6a /bin/bash
> rpc error: code = 2 desc = oci runtime error: exec failed: container_linux.go:1153: sending signal 0 to pid 23887 caused "permission denied"

**Comment by lukaszgryglicki** [comment]
Yeah I have it too, any solutions?

**Comment by olegantonyan** [comment]
Same here
```Client:
 Version:      17.07.0-ce
 API version:  1.31
 Go version:   go1.8.5
 Git commit:   87847530f717
 Built:        Tue Nov  7 12:00:00 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.07.0-ce
 API version:  1.31 (minimum version 1.12)
 Go version:   go1.8.5
 Git commit:   87847530f717
 Built:        Tue Nov  7 12:00:00 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by malanve** [comment]
Same here

Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.6
Storage Driver: overlay2
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: oci runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 4.14.4-1-default
Operating System: openSUSE Tumbleweed
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 62.81 GiB
Name: virts
ID: 5T6Y:AWTF:2JIX:DPQX:SQS4:FE7L:QMF3:NAZS:4DJW:JPWE:7TUV:SYP2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8

sudo docker exec -it 0e81a20f20d5 bash
rpc error: code = 13 desc = invalid header field value "oci runtime error: exec failed: container_linux.go:1156: sending signal 0 to pid 2530 caused \"permission denied\"\n"

Any news on this?

**Comment by GuzelGarifullina** [comment]
Have the same issue. Everything had worked fine before my system hung and I had no choice but to make a forced reboot. Reinstallation of docker also didn't help. 

Looks like this issue is connected  with #27742 in some way. 

For me answer by [vtutrinov](https://github.com/vtutrinov) fixed the issue:

Run docker with
```--security-opt apparmor:unconfined```

**Comment by lukaszgryglicki** [comment]
For me this kinda worked but....
I was facing many many strange issues, at first glance non-related to docker, but after few weeks struggle, I've installed multiple native Grafanas instead of dockerized, and all problems go away.

I needed to disable apparmour and change storage driver from overlay2 to aufs.
Docker version was the same on two different servers, Linux version the same, but different kernels - so maybe Linux kernel was the cause?
